compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ImmutableTypeCloner.java

```diff
@@ -542,14 +542,13 @@ private static void populateImmutableStructureFields(Types types, SymbolTable sy
 
     private static String getPackageAlias(SymbolEnv env, String compUnitName, PackageID typePkgId) {
         for (BLangImportPackage importStmt : env.enclPkg.imports) {
-            if (!typePkgId.equals(importStmt.symbol.pkgID)) {
+            if (!importStmt.compUnit.value.equals(compUnitName)) {
                 continue;
             }
 
-            if (importStmt.compUnit.value.equals(compUnitName)) {
+            if (importStmt.symbol != null && typePkgId.equals(importStmt.symbol.pkgID)) {
```

Comment:
Any particular reason to swap the order of the two checks?

Reply:
As mentioned by @rdhananjaya we can early exit by checking the equality of comUnitName